html/semantics/forms/resetting-a-form/reset-form-event-realm.html

```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
+<title>reset() event firing realm</title>
+<link rel="help" href="https://html.spec.whatwg.org/#resetting-a-form">
+<link rel="help" href="https://dom.spec.whatwg.org/#concept-event-fire">
+<link rel="author" title="Domenic Denicola" href="mailto:d@domenic.me">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+
+<iframe src="reset-form-event-realm-support.html"></iframe>
+<iframe></iframe>
+
+<script>
+"use strict";
+
+window.onload = () => {
+  const frame0Form  = frames[0].document.forms[0];
+  const frame1Body = frames[1].document.body;
+
+  frame1Body.appendChild(frame0Form);
+
+  let resetCalled = false;
+  frame0Form.onreset = ev => {
+    resetCalled = true;
+
+    assert_equals(ev.constructor, frames[0].Event, "the event must be created in the realm of the target");
```

Comment:
This is making assumptions about "the Realm of target" which are based on not-yet-decided spec text, no?

I would prefer it if you actually compared to the Realm of target, not to some Realm that may or may not actually be the Realm of target....

The question is how to extract the actual Realm from the target, right?  I don't think we have anything that allows us to do that, except observations like this...  In particular, the Realm of the target may or may not match the Realm of its prototype, and you can't tell short of observing the Realm of the target indirectly via things created off the target or something.

Perhaps it would make sense to require that the Realm of target in fact matches its proto chain in cases when no manual proto sets are being done in the script.  In that case we should be able to compare the Realms of any two objects by comparing Object.getPrototypeOf(obj.constructor), right?